backend/e2etests/articles.test.js

```diff
@@ -0,0 +1,222 @@
+    });
+
+    it("returns 400 status when Tenant is not set", async function () {
+      return request("http://localhost:9999/api")
+        .get(`/${baseEndpoint}`)
+        .expect(400);
+    });
+  });
+
+  describe("POST", function () {
+    const baseEndpoint = "articles";
+
+    it("returns 200 status", async function () {
+      return request("http://localhost:9999/api")
+        .post(`/${baseEndpoint}`)
+        .send({
+          name: "Oss Stickers",
+          description: "Stickers for Oss open source projects",
+          quantity: 10,
+          price: 5,
+          pictures: [
+            {
+              id: 1,
+              alt_text: "oss cameroon sticker image",
+            },
+          ],
+          metadata: "string",
+        })
+        .set("Tenant", "1")
+        .expect(200)
+        .expect("Content-Type", "application/json; charset=utf-8");
+    });
+
+    it("returns 400 status when Tenant is not set", async function () {
+      return request("http://localhost:9999/api")
+        .post(`/${baseEndpoint}`)
+        .expect(400);
+    });
+  });
+});
+
+describe("articles/{id}", () => {
+  const baseEndpoint = "articles";
+
+  describe("GET", function () {
+    it("returns 200 status", async function () {
+      return request("http://localhost:9999/api")
+        .get(`/${baseEndpoint}/1`)
+        .set("Tenant", "1")
+        .expect(200)
+        .expect("Content-Type", "application/json; charset=utf-8");
+    });
+
+    it("returns 400 status when Tenant is not set", async function () {
+      return request("http://localhost:9999/api")
+        .get(`/${baseEndpoint}/1`)
+        .expect(400);
+    });
+
+    it("returns 404 status when the resource is not found", async function () {
+      //Please unskip this test once support for this error status is done
+      this.skip();
+      return request("http://localhost:9999/api")
+        .get(`/${baseEndpoint}/404`)
+        .set("Tenant", "1")
+        .expect(404);
+    });
+  });
+
+  describe("DELETE", function () {
+    const baseEndpoint = "articles";
+    let article_id;
+    before(function () {
+      const data = {
+        name: "data",
+        description: "data description",
+        quantity: 10,
+        price: 2,
+        pictures: [
+          {
+            id: 2,
+            alt_text: "data picture",
+          },
+        ],
+        metadata: "string",
+      };
+
+      return axios({
+        url: `http://localhost:9999/api/${baseEndpoint}`,
+        method: "post",
+        data: data,
+        headers: { Tenant: "1" },
+      }).then(function (res) {
+        article_id = res.data.id;
+        return res;
+      });
+    });
+
+    it("returns 200 status", async function () {
+      return request("http://localhost:9999/api")
+        .delete(`/${baseEndpoint}/${article_id}`)
+        .set("Tenant", "1")
+        .expect(200)
+      });
+
+    it("returns 400 status when Tenant is not set", async function () {
+      this.skip()
```

Comment:
Skipped tests are tests that are present in the api spec but not yet implemented or are simply broken. Meaning at some point these tests should be enabled.